var/spack/repos/spack_repo/builtin/packages/quandary/package.py

```diff
@@ -0,0 +1,67 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Quandary(CachedCMakePackage, CudaPackage, ROCmPackage):
+    """Optimal control for open quantum systems"""
+
+    homepage = "https://github.com/LLNL/quandary"
+    git = "https://github.com/LLNL/quandary.git"
+
+    maintainers("steffi7574", "tdrwenski")
+
+    license("MIT", checked_by="tdrwenski")
+
+    version("main", branch="main", preferred=True)
+    version("learning", branch="learning")
+
+    depends_on("cxx", type="build")
+
+    depends_on("petsc~hypre~metis~fortran")
+    depends_on("petsc+debug", when="+debug")
+    depends_on("slepc", when="+slepc")
+
+    depends_on("blt", type="build")
+    conflicts("^blt@:0.3.6", when="+rocm")
+
+    with when("+rocm"):
+        depends_on("petsc+rocm")
+        for arch_ in ROCmPackage.amdgpu_targets:
+            depends_on(
+                "petsc amdgpu_target={0}".format(arch_), when="amdgpu_target={0}".format(arch_)
+            )
```

Comment:
I believe you can tighten this up a bit:

```suggestion
        for arch_ in ROCmPackage.amdgpu_targets:
            depends_on(
                f"petsc+rocm amdgpu_target={arch_}", when=f"amdgpu_target={arch_}"
            )
```

Reply:
Fixed in 00982c11029